lib/kubernetes-deploy/kubernetes_resource/customresourcedefinition.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class CustomResourceDefinition < KubernetesResource
+    TIMEOUT = 10.seconds
+    PREDEPLOY = true
+
+    def sync
+      _, _err, st = kubectl.run("get", kind, @name)
+      @status = st.success? ? "Available" : "Unknown"
+      @found = st.success?
+    end
+
+    def deploy_succeeded?
+      exists?
+    end
+
+    def deploy_failed?
```

Comment:
Might make sense to look at the `NamesAccepted` condition here (see `waitCRDReady` in cloudbuddies)